src/classes/keyboard.class.js

```diff
@@ -152,6 +173,603 @@ class Keyboard {
+            }
+        };
+        let addMacron = (char) => {
+            switch (char) {
+            case "a":
+                return "ā";
+            case "A":
+                return "Ā";
+            case "e":
+                return "ē";
+            case "E":
+                return "Ē";
+            case "g":
+                return "ḡ";
+            case "G":
+                return "Ḡ";
+            case "i":
+                return "ī";
+            case "I":
+                return "Ī";
+            case "o":
+                return "ō";
+            case "O":
+                return "Ō";
+            case "u":
+                return "ū";
+            case "U":
+                return "Ū";
+            case "y":
+                return "ȳ";
+            case "Y":
+                return "Ȳ";
+            case "é":
+                return "ḗ";
+            case "É":
+                return "Ḗ";
+            case "è":
+                return "ḕ";
+            case "È":
+                return "Ḕ";
+            default:
+                return char;
+            }
+        };
+        let addCedilla = (char) => {
+            switch (char) {
+            case "c":
+                return "ç";
+            case "C":
+                return "Ç";
+            case "d":
+                return "ḑ";
+            case "D":
+                return "Ḑ";
+            case "e":
+                return "ȩ";
+            case "E":
+                return "Ȩ";
+            case "g":
+                return "ģ";
+            case "G":
+                return "Ģ";
+            case "h":
+                return "ḩ";
+            case "H":
+                return "Ḩ";
+            case "k":
+                return "ķ";
+            case "K":
+                return "Ķ";
+            case "l":
+                return "ļ";
+            case "L":
+                return "Ļ";
+            case "n":
+                return "ņ";
+            case "N":
+                return "Ņ";
+            case "r":
+                return "ŗ";
+            case "R":
+                return "Ŗ";
+            case "s":
+                return "ş";
+            case "S":
+                return "Ş";
+            case "t":
+                return "ţ";
+            case "T":
+                return "Ţ";
+            default:
+                return char;
+            }
+        };
+        let addOverring = (char) => {
+            switch (char) {
+            case "a":
+                return "å";
+            case "A":
+                return "Å";
+            case "u":
+                return "ů";
+            case "U":
+                return "Ů";
+            case "w":
+                return "ẘ"; // capital w with overring not supported on bépo layout apparently
+            case "y":
+                return "ẙ"; // same for capital y with overring
+            default:
+                return char;
+            }
+        };
+        let toGreek = (char) => {
```

Comment:
I don't believe adding a modifier is the right way to handle different alphabets. You should define them directly in the keyboard file, in `*_cmd` fields. See my review on the BEPO layout file